modules/infra/submodules/storage/main.tf

```diff
@@ -24,6 +24,18 @@ locals {
       policy_json = data.aws_iam_policy_document.costs[0].json
       arn         = aws_s3_bucket.costs[0].arn
     } : {}
+    flyte_metadata = var.flyte_enabled ? {
```

Comment:
@miguelhar It's now unclear to me if references here should be `var.flyte_enabled` or `var.storage.flyte_enabled`. If it's the former then it doesn't seem like we need the `flyte_enabled` key in the storage module at all. If it's the latter then I'm unsure how we use the value of `var.flyte_enabled` to set the value of `var.storage.flyte_enabled`.